libs/vkd3d/shaders/cs_workgraph_distribute_workgroups.comp

```diff
@@ -0,0 +1,204 @@
+#version 450
+#extension GL_KHR_shader_subgroup_basic : require
+#extension GL_KHR_shader_subgroup_ballot : require
+#extension GL_KHR_shader_subgroup_arithmetic : require
+#extension GL_KHR_shader_subgroup_vote : require
+#extension GL_EXT_buffer_reference : require
+#extension GL_GOOGLE_include_directive : require
+
+layout(local_size_x_id = 0) in;
+layout(constant_id = 1) const uint FUSED_DIVIDER = 0;
+layout(constant_id = 2) const bool COMPACT_BROADCAST_NODES = false;
+
+// If maxComputeWorkGroups[0] is huge, we don't have to worry.
+// AMD and NV support that at least.
+// When we introduce mesh shaders, we'll have to be more conservative.
+// However, mesh nodes will require some esoteric MDI handling anyway, so ... eh.
+layout(constant_id = 3) const bool REQUIRE_WG_DIVIDER = true;
+
+#include "cs_workgraph_data_structures.h"
+
+layout(buffer_reference, buffer_reference_align = 8, std430) buffer NodeAtomics
+{
+	uint payload_atomic;
+	uint fused_atomic;
+	uint node_counts[];
+};
+
+// Sharing across threads requires coherent, even within a subgroup.
+layout(buffer_reference, buffer_reference_align = 16, std430) coherent buffer IndirectCommandsBuffer
```

Comment:
`subgroupcoherent` is a thing, although for that to do anything we'd also need to opt-into vulkan memory model for these shaders (`#pragma use_vulkan_memory_model`).

Might be worth looking into so drivers don't pessimize code gen too much.